File: examples/nodejs/kafka-producer-consumer/consumer.js
import { Kafka } from 'kafkajs';

const brokers = (process.env.KAFKA_BROKERS || 'localhost:9092').split(',');
const kafka = new Kafka({ clientId: 'node-consumer', brokers });
const consumer = kafka.consumer({ groupId: 'node-group-1' });

await consumer.connect();
await consumer.subscribe({ topic: 'app-events', fromBeginning: true });

await consumer.run({ eachMessage: async ({ message }) => console.log('[consumer] got', message.value?.toString()) });

File: examples/python/kafka-producer-consumer/consumer.py
import os
from kafka import KafkaConsumer

brokers = os.getenv("KAFKA_BROKERS", "localhost:9092").split(",")
consumer = KafkaConsumer(
    "app-events",
    bootstrap_servers=brokers,
    group_id="python-group-1",
    auto_offset_reset="earliest",
)

print("[consumer] waiting...")
for msg in consumer:
    print("[consumer] got", msg.value.decode())
    break
